stash/commits.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2020 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stash
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"reflect"
+
+	"github.com/go-logr/logr"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	"github.com/fluxcd/go-git-providers/httpclient"
+)
+
+var (
+	ErrorGetCommitMultipleItems = errors.New("multiple items returned for commit name")
+)
+
+const (
+	CommitsURI = "commits"
+)
+
+type Commit struct {
+	SessionInfo        `json:"sessionInfo,omitempty"`
+	Author             User   `json:"author,omitempty"`
+	AuthorTimestamp    int64  `json:"authorTimestamp,omitempty"`
+	Committer          User   `json:"committer,omitempty"`
+	CommitterTimestamp int64  `json:"committerTimestamp,omitempty"`
+	DisplayId          string `json:"displayId,omitempty"`
```

Comment:
I suspect this should be (following both Go conventions, and the other names in your struct)`DisplayID`